Lombiq.ChartJs.Samples/Indexes/ExpensePartIndex.cs

```diff
@@ -0,0 +1,4 @@
+namespace Lombiq.ChartJs.Samples.Indexes;
+public class ExpensePartIndex : ItemPartIndex
```

Comment:
Could you please provide some sample code for this?

Reply:
For example instead of 
```csharp
(await _session.QueryIndex<ExpensePartIndex>()
    .ListAsync())
    .Where(expense => expense.Amount is not null)
    .Select(expense => decimal.ToDouble(expense.Amount.Value))
    .Sum() as double?,
```
you can do
```csharp
(await _session.QueryIndex<NumericPartIndex>(index =>
        index.Published &&
        index.Latest &&
        index.ContentType == ContentTypes.Expense &&
        index.ContentPart == nameof(ExpensePart) &&
        index.ContentField == nameof(ExpensePart.Amount) &&
        index.Numeric != null)
    .ListAsync())
    .Select(expense => decimal.ToDouble(index.Numeric))
    .Sum() as double?,
```

You probably have to enable the _OrchardCore.ContentFields.Indexing.SQL_ feature.

Reply:
This solution looks good to collect data for _"Balance"_, but as I see, it generates index tables for all the parts comes from _OrchardCore.ContentFields_, and indexes all of it. This can cause too mutch overhead in a real application, but it absolutely depends of the application.
Can we configure the indexing behavior? I mean is it possible to select the content types and the attached content parts for indexing.

I think, making custom indexes instead of using _*PartIndex_ from _OrchardCore.ContentFields_ is most optimal and cleaner on the query side in case of _History_.

Reply:
In a sample project we should minimize "infrastructure" code, to make the demo reading experience easier. So cutting out all  the unnecessary migrations and index providers makes the project overall cleaner

In real projects if you add a field you will almost certainly want to filter by it eventually. But if you only add what you want to filter by now, or if you want to add new fields later you have a problem. Adding columns to a custom index after it's already published is a huge pain, becuase you need migration and you have to trigger re-indexing on the existing items too. Also I don' think the overhead is that big unless you need to handle huge tabular datasets. But in that scenario you probably have custom display logic and want to use a part with primitive properties instead of fields, to cut out on the field overhead in general so index generation wouldn't be an problem.

In the end I think the development and maintenance costs of using custom indexes outweigh their performance gain except in very specific circumstances.